src/jit/lsraarmarch.cpp

```diff
@@ -711,6 +711,13 @@ void Lowering::TreeNodeInfoInitPutArgStk(GenTreePutArgStk* argNode, fgArgTabEntr
             MakeSrcContained(argNode, putArgChild);
         }
     }
+#ifdef _TARGET_ARM_
+    else if (putArgChild->TypeGet() == TYP_DOUBLE) {
+        // Even though TYP_DOUBLE uses 2 numSlots, we consume 1 srcCount in TYP_DOUBLE
+        argNode->gtLsraInfo.srcCount = 1;
+        putArgChild->SetContained();
```

Comment:
Looks like `argNode->gtLsraInfo.srcCount = 1;` is unnecessary, since that was already done above.

Why do `TYP_DOUBLE` nodes need to get special treatment, marked as contained?

Is this issue really a matter of loosening the assert below, maybe changing it to:
```
#ifdef _TARGET_ARM_
        // We must not have a multi-reg struct; double uses 2 slots and isn't a multi-reg struct.
        assert((info->numSlots == 1) || ((info->numSlots == 2) && (putArgChild->TypeGet() == TYP_DOUBLE)));
#else // !_TARGET_ARM_
        // We must not have a multi-reg struct
        assert(info->numSlots == 1);
#endif // !_TARGET_ARM_
```